decoder/src/frame.rs

```diff
@@ -160,12 +212,52 @@ fn format_args_real(
                 let micros = x % 1_000_000;
                 write!(buf, "{}.{:06}", seconds, micros)?;
             }
+            Some(DisplayHint::Bitflags {
+                name,
+                package,
+                disambiguator,
+            }) => {
+                // The bitflags hint is only used internally, in `Format` impls generated by
+                // `defmt::bitflags!`.
+                let key = BitflagsKey {
+                    ident: name.clone(),
+                    package: package.clone(),
+                    disambig: disambiguator.clone(),
+                };
+                match self.table.bitflags.get(&key) {
+                    Some(flags) => {
+                        // FIXME flag print order does not match definition order
+                        // (does some part of the toolchain alphabetically sort them?)
```

Comment:
interesting observation: perhaps the linker sorts the symbols alphabetically within a section? maybe we could prefix the symbols (`00_$first`, `01_$second`, etc.) so that even after sorting alphabetically they stay in the declaration order? that doesn't have to be addressed in this PR but we should create an issue for this